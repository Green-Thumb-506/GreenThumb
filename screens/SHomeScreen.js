import {
  StyleSheet,
  Text,
  View,
  ScrollView,
  FlatList,
  TouchableOpacity,
  Image,
} from 'react-native';

import React from 'react';
import Icon from 'react-native-vector-icons/Ionicons';
import {
  fetchPlantDB,
  fetchUserPlants,
  removePlantUserDB,
} from '../services/Api.js';
import { withNavigation } from 'react-navigation'

type State = {
  plants: any,
  userPlants: any,
  refreshing: boolean,
}

export default class SHomeScreen extends React.Component {
  state: State = {
    plants: null,
    userPlants: null,
    refreshing: false,
  }

  componentDidMount() {
    this._fetchPlants();
  }

  static navigationOptions = {
      title: 'My Garden',
      headerStyle: { backgroundColor: '#3BAD87' },
      headerTitleStyle: { fontSize: 30, height: 60, paddingTop: 10, },
  }

  render() {
      return (
          <View style={styles.container}>
            <View style={{flex: 1}}>
               {this.state.plants && this._renderFlatList()}
            </View>
          </View>
      );
  }

  _renderFlatList() {
    if (!this.state.plants) { return null; }

    return (
       <FlatList
          data={this.state.plants}
          renderItem={this._renderItem}
          keyExtractor={this._keyExtractor}
          ItemSeparatorComponent={this._renderCellSeperator}
          onRefresh={this._fetchPlants}
          refreshing={this.state.refreshing}
       />
    );
   }

   _renderCellSeperator() {
      return (
        <React.Fragment>
          <View style={styles.cellSeperator} />
        </React.Fragment>
      );
    }


   _renderItem = (item: any, index: number) => {
     const name = item.item.name || "";
     const imageUri = item.item.picture || "";
     const key = item.item.key || "";

     return (
       <TouchableOpacity
         onPress={() => {
           /* istanbul ignore next */
           this._onPressPlant(item.item);
         }}
         key={item.id}
         style={styles.item}
       >
         <Image
           style={styles.image}
           source={{uri: imageUri}}
         />
         <View style={styles.textView}>
           <Text style={{ fontSize: 18, fontWeight: 'bold' }}>{name}</Text>
         </View>
         <TouchableOpacity
          onPress={() => {
            this._removePlant(key);
          }}
         >
            <Icon size={45} name={'ios-trash'} />
         </TouchableOpacity>
      </TouchableOpacity>
     );
   }
   /* istanbul ignore next */

   _removePlant = async (name: string) => {
     var updatedPlantDB = await removePlantUserDB(name);
     this._fetchPlants();
   }

   /* istanbul ignore next */
   _onPressPlant = (item: any) => {
     this.props.navigation.navigate('DetailedPlant', {
       plant: item,
     });
   }

   /* istanbul ignore next */
   _keyExtractor = (item, index) => {
     return index.toString();
   }

   /* istanbul ignore next */
   _fetchPlants = async (): void => {
     var plantDb = await fetchPlantDB();
     var userPlants = await fetchUserPlants();
     this.setState({
       userPlants: userPlants || {},
     });
     var plantsArr = [];
     for (var key in userPlants) {
      if (userPlants[key] && plantDb.plantDictionary[key]) {
        plantsArr.push(plantDb.plantDictionary[key]);
      }
     }
     this.setState({
       plants: plantsArr || [],
     });
   }
}

/* istanbul ignore next */
const styles = StyleSheet.create({
    container: {
      flex: 1,
      paddingTop: 10,
    },
    cellSeperator: {
      height: 5,
      width: '100%',
    },
    image: {
      height: 80,
      width: 80,
      margin: 10,
      borderRadius: 40,
    },
    item: {
        flex: 1,
        height: 100,
        backgroundColor: '#bdc3c7',
        flexDirection: 'row',
        justifyContent: 'space-between',
        alignItems: 'center',
        padding: 10,
        fontWeight: 'bold',
        borderWidth: 3,
        marginLeft: 10,
        marginRight: 10,
     },
     textView: {
        width: '60%',
     },
});
