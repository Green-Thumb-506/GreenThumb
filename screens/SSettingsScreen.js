import React from 'react';
import { Text, View, StyleSheet } from 'react-native';

export default class SSettingsScreen extends React.Component {

    static navigationOptions = {
        title: 'Settings',
        headerStyle: { backgroundColor: '#3BAD87' },
        headerTitleStyle: { fontSize: 40, height: 60 },
    }

    render() {
        return (
            <View style={styles.container}>
            </View>
        );
    }

}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        position: 'absolute',
        justifyContent: 'center',
        alignItems: 'center',
        backgroundColor: '#F5FCFF'
    },
});
