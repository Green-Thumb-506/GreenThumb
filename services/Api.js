import Constants from 'expo-constants';

var installationId = Constants.installationId;
var installationIdHouse = Constants.installationId + "house";

var email = "";
var name = "";
var realtimeDBname = "";

export const loginUser = async (userName: string, password: string): Promise<Object> => {
  var data = {
    email: userName,
    password: password,
    returnSecureToken: true
  };
  const jsonData = JSON.stringify(data);
  let request = {
      method: 'POST',
      body: jsonData,
      'Content-Type': 'application/json'
  }
  try {
    const res = await fetch('https://identitytoolkit.googleapis.com/v1/accounts:signInWithPassword?key=AIzaSyD7JOQOxyW82ZPpDYDjaJNwR7MyoxfibM8', request);
    const json = await res.json();
    email = userName;
    return json;
  } catch (err) {
    // console.error("#25 Login error: ", err);
    return null;
  }
}

export const signUp = async (userName: string, password: string): Promise<Object> => {
  var data = {
    email: userName,
    password: password,
    returnSecureToken: true
  };
  const jsonData = JSON.stringify(data);
  let request = {
      method: 'POST',
      body: jsonData,
      'Content-Type': 'application/json'
  }
  try {
    const res = await fetch('https://identitytoolkit.googleapis.com/v1/accounts:signUp?key=AIzaSyD7JOQOxyW82ZPpDYDjaJNwR7MyoxfibM8', request);
    const json = await res.json();
    return json;
  } catch (err) {
    console.error("#25 Sign Up error: ", JSON.stringify(err));
    return null;
  }
}

export const getHouses = async (): Promise<Object> => {
  try {
    const res = await fetch('https://spike-exercise-f7fec.firebaseio.com/.json');
    const json = await res.json();
    console.log("JSON: " + JSON.stringify(json))
    return json;
  } catch (err) {
    console.error("#25 Sign Up error: ", err);
    return null;
  }
}


export const storeUser = async (userName: string, userEmail: string) => {
  email = userEmail;
  name = userName;

  var data = {
    email: userEmail,
    name: userName,
  };
  const jsonData = JSON.stringify(data);
  console.log(jsonData)
  let request = {
      method: 'PUT',
      body: jsonData,
      'Content-Type': 'application/json'
  }
  try {
    const res = await fetch('https://spike-exercise-f7fec.firebaseio.com/' + [installationId] + '.json', request);
    const json = await res.json();
    console.log("JSON: " + JSON.stringify(json))
    realtimeDBname = json && json.name;
    return json;
  } catch (err) {
    console.error("#28 Sign Up error: ", err);
    return null;
  }
}

export const getUserName = async (): String => {
  try {
    const res = await fetch('https://spike-exercise-f7fec.firebaseio.com/.json');
    const json = await res.json();
    return json && json[installationId] && json[installationId].name || "";
  } catch (err) {
    return "Name: " + name;
  }
}

export const getNumRoommates = async (): number => {
  try {
    const res = await fetch('https://spike-exercise-f7fec.firebaseio.com/.json');
    const json = await res.json();
    return json && json[installationIdHouse] && json[installationIdHouse].numRoommates || 0;
  } catch (err) {
    // console.error("#28 Sign Up error: ", err);
    return 0;
  }
}

export const getPrice = async (): number => {
  try {
    const res = await fetch('https://spike-exercise-f7fec.firebaseio.com/.json');
    const json = await res.json();
    return json && json[installationIdHouse] && json[installationIdHouse].price || 0;
  } catch (err) {
    return 0;
  }
}

export const addListing = async (price: string, numRoommates: string, name: string, email: string) => {
  var data = {
    price: price,
    numRoommates: numRoommates,
    name: name,
    email: email,
  };
  const jsonData = JSON.stringify(data);
  let request = {
      method: 'PUT',
      body: jsonData,
      'Content-Type': 'application/json'
  }

  try {
    const res = await fetch('https://spike-exercise-f7fec.firebaseio.com/' + [installationIdHouse] + '.json', request);
    const json = await res.json();
    console.log("JSON: " + JSON.stringify(json))
    return json;
  } catch (err) {
    return null;
  }
}

export const addFavorite = async (key: string) => {
  var data = {
    favorite: true,
  };
  const jsonData = JSON.stringify(data);
  let request = {
      method: 'PATCH',
      body: jsonData,
      'Content-Type': 'application/json'
  }
  try {
    const res = await fetch('https://spike-exercise-f7fec.firebaseio.com/' + key + '/favorites/' +  [installationId] + '.json', request);
    const json = await res.json();
    console.log("JSON: " + JSON.stringify(json))
    return json;
  } catch (err) {
    return null;
  }
}

export const getUserEmail = (): String => {
  return email;
}

export const isLoggedIn = (): boolean => {
  if (email) {
    return true;
  }
  return false
}
