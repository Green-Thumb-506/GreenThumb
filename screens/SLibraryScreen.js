import {
  StyleSheet,
  Text,
  View,
  ScrollView,
  FlatList,
  TouchableOpacity,
  Image,
} from 'react-native';

import React from 'react';
import Icon from 'react-native-vector-icons/Ionicons';
import { fetchPlantDB, addPlantUserDB } from '../services/Api.js';

type State = {
  plants: any,
  refreshing: boolean,
}

export default class SLibraryScreen extends React.Component {
  state: State = {
    plants: null,
    refreshing: false,
  }

  componentDidMount() {
    this._fetchPlants();
  }

  static navigationOptions = {
      title: 'Library',
      headerStyle: { backgroundColor: '#C85B6C' },
      headerTitleStyle: { fontSize: 40, height: 60 },
  }

  render() {
    return (
      <View style={styles.container}>
        <View style={{flex: 1}}>
           {this.state.plants && this._renderFlatList()}
        </View>
      </View>
    );
  }

  /* istanbul ignore next */
  _renderFlatList() {
    if (!this.state.plants || !this.state.plants.plantDictionary) { return null; }
    const plantDictionary = this.state.plants.plantDictionary;
    var data = [];

    for (var i in plantDictionary)
        data.push(plantDictionary[i]);
    return (
       <FlatList
          data={data}
          renderItem={this._renderItem}
          keyExtractor={this._keyExtractor}
          ItemSeparatorComponent={this._renderCellSeperator}
          onRefresh={this._fetchPlants}
          refreshing={this.state.refreshing}
       />
    );
 }

 /* istanbul ignore next */
 _renderCellSeperator() {
    return (
      <React.Fragment>
        <View style={styles.cellSeperator} />
      </React.Fragment>
    );
  }

  /* istanbul ignore next */
 _renderItem = (item: any, index: number) => {
   const name = item.item.name || "";
   const imageUri = item.item.picture || "";
   const key = item.item.key || "";

   return (
     <TouchableOpacity
       onPress={() => this._onPressPlant(item.item)}
       key={item.id}
       style={styles.item}
     >
       <Image
         style={styles.image}
         source={{uri: imageUri}}
       />
       <View style={styles.textView}>
         <Text style={{ fontSize: 18, fontWeight: 'bold' }}>{name}</Text>
       </View>
       <TouchableOpacity
        onPress={() => {
          this._addPlant(key);
        }}
       >
          <Icon size={45} name={'ios-add'} />
       </TouchableOpacity>
    </TouchableOpacity>
   );
 }

 /* istanbul ignore next */
 _addPlant = async (name: string) => {
   await addPlantUserDB(name);
 }

 /* istanbul ignore next */
 _onPressPlant = (item: any) => {
   this.props.navigation.navigate('DetailedPlant', {
     plant: item,
   });
 }

 /* istanbul ignore next */
 _keyExtractor = (item, index) => {
   return index.toString();
 }

 /* istanbul ignore next */
 _fetchPlants = async (): void => {
    var res = await fetchPlantDB();
    this.setState({
      plants: res || {},
    })
 }


}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    paddingTop: 10,
  },
  cellSeperator: {
    height: 5,
    width: '100%',
  },
  image: {
    height: 80,
    width: 80,
    margin: 10,
    borderRadius: 40,
  },
  item: {
      flex: 1,
      height: 100,
      backgroundColor: '#bdc3c7',
      flexDirection: 'row',
      justifyContent: 'space-between',
      alignItems: 'center',
      padding: 10,
      fontWeight: 'bold',
      borderWidth: 3,
      marginLeft: 10,
      marginRight: 10,
   },
   textView: {
      width: '60%',
   },
});
