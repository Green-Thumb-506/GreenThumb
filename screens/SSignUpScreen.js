"use strict";

import React from 'react';
import {
  StyleSheet,
  Text,
  View,
  Image,
  Alert,
  ScrollView,
} from 'react-native';
import { Input, Button } from 'react-native-elements';
import { signUp, storeUser } from '../services/Api.js';

/* istanbul ignore next */
type State = {
  email: string,
  password: string,
  name: string,
};

class SSignUpScreen extends React.PureComponent<Props, State> {

  /* istanbul ignore next */
  state: State = {
    email: '',
    password: '',
    name: '',
  };

  /* istanbul ignore next */
  static navigationOptions = {
    title: 'Create an Account',
    headerStyle: { backgroundColor: '#bdc3c7' },
    headerTitleStyle: { color: '#2c3e50', },
  }

  /* istanbul ignore next */
  render(): React$Element<any> {
    return (
      <ScrollView contentContainerStyle={styles.container}>
        <Image
          style={styles.image}
          source={require('../images/green-tea.png')}
        />
        <Input
          placeholder='John Deer'
          containerStyle={styles.inputEmail}
          label="Name"
          onChangeText={text => {
            /* istanbul ignore next */
            this._onChangeName(text)
          }}
        />
        <Input
          placeholder='email@address.com'
          containerStyle={styles.inputEmail}
          label="Email"
          onChangeText={text => {
            /* istanbul ignore next */
            this._onChangeEmail(text)
          }}
        />
        <Input
          placeholder='Password'
          containerStyle={styles.input}
          secureTextEntry={true}
          label="Password"
          onChangeText={text => {
            /* istanbul ignore next */
            this._onChangePassword(text)
          }}
        />
        <Button
          title="Create Account"
          style={styles.button}
          buttonStyle={{
            backgroundColor:'#2c3e50',
          }}
          onPress={this._proceedLogin}
        />
      </ScrollView>
    );
  }

  /* istanbul ignore next */
  _onChangeName = (text: string) => {
    this.setState({
      name: text,
    });
  }

  /* istanbul ignore next */
  _onChangeEmail = (text: string) => {
    this.setState({
      email: text,
    });
  }

  /* istanbul ignore next */
  _onChangePassword = (text: string) => {
    this.setState({
      password: text,
    });
  }

  /* istanbul ignore next */
  _proceedLogin = async () => {
    if (!this.state.email || !this.state.password || !this.state.name) {
      Alert.alert(
        'Error!',
        'Please enter your Name, Email, and Password!',
        [
          {text: 'OK', onPress: () => console.log('OK Pressed')},
        ],
        {cancelable: false},
      );
      return;
    }

    var res = await signUp(this.state.email, this.state.password);
    if (!res || res.error) {
      Alert.alert(
        'Error!',
        res && res.error && res.error.message || 'There was an error signing up!',
        [
          {text: 'OK', onPress: () => console.log('OK Pressed')},
        ],
        {cancelable: false},
      );
      return;
    }
    this.props.navigation.navigate('Main');
  }
}

/* istanbul ignore next */
const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#bdc3c7',
    alignItems: 'center',
    justifyContent: 'flex-start',
    paddingTop: 50,
  },
  image: {
    height: 120,
    width: 120,
    marginBottom: 60,
  },
  inputStyle: {
    paddingBottom: 100,
  },
  headerText: {
    marginBottom: 30,
    fontWeight: '500',
    fontSize: 20,
    color: '#2c3e50',
  },
  button: {
    width: '70%',
    color:  '#2c3e50',
    paddingBottom: 30,
  },
  input: {
    width: '70%',
    paddingBottom: 20,
  },
  inputEmail: {
    width: '70%',
    paddingBottom: 10,
  },
});

module.exports = SSignUpScreen;
