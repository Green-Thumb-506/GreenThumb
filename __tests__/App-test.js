import React from 'react';
import NavigationTestUtils from 'react-navigation/NavigationTestUtils';
import renderer from 'react-test-renderer';

import App from '../App';
import SHomeScreen from '../screens/SHomeScreen.js';
import SSettingsScreen from '../screens/SSettingsScreen.js';
import SLibraryScreen from '../screens/SLibraryScreen.js';
import SSignUpScreen from '../screens/SSignUpScreen.js';

import { signUp, loginUser, fetchPlantDB } from '../services/Api.js';

jest.mock('expo', () => ({
  AppLoading: 'AppLoading',
}));

jest.mock('../navigation/AppNavigator', () => 'AppNavigator');

describe('App', () => {
  jest.useFakeTimers();

  beforeEach(() => {
    NavigationTestUtils.resetInternalState();
  });

  it(`renders the loading screen`, () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it(`renders the root without loading screen`, () => {
    const tree = renderer.create(<App skipLoadingScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it(`renders the loading screen`, () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('SHomeScreen', () => {
    describe('Rendering', () => {
        it('should match to snapshot', () => {
            const navigation = { navigate: jest.fn() };
            expect(true)
        });
    });
});

describe('SLibraryScreen', () => {
  jest.useFakeTimers();

  beforeEach(() => {
    NavigationTestUtils.resetInternalState();
  });

  it(`renders the settings screen`, () => {
    const tree = renderer.create(<SLibraryScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('SSignUpScreen', () => {
  jest.useFakeTimers();

  beforeEach(() => {
    NavigationTestUtils.resetInternalState();
  });

  it(`renders the sign-up screen`, () => {
    const tree = renderer.create(<SSignUpScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('API', async () => {
  jest.useFakeTimers();

  beforeEach(() => {
    NavigationTestUtils.resetInternalState();
  });

  it(`Testing Sign Up Non Async`, () => {
    const json = signUp("username", "password")
    expect(json);
  });

  it(`Testing Log In Non Async`, () => {
    const json = loginUser("username", "password")
    expect(json);
  });

  it(`Testing Fetch DB Non Async`, () => {
    const json = fetchPlantDB()
    expect(json);
  });

  it(`Testing LogIn Async`, () => {
    return loginUser().then(data => {
      expect(data).toBe(data);
    });
  });

  it(`Testing FetchDB Async`, () => {
    return fetchPlantDB().then(data => {
      expect(data).toBe(data);
    });
  });
});

test('SignUp Async', async () => {
  fetch = jest.fn(() => Promise.resolve());
  await expect(signUp("username", "password")).resolves.toBe(null);
});

test('LogIn Async', async () => {
  fetch = jest.fn(() => Promise.resolve());
  await expect(loginUser("username", "password")).resolves.toBe(null);
});

test('FetchDB Async', async () => {
  fetch = jest.fn(() => Promise.resolve());
  await expect(fetchPlantDB()).resolves.toBe(null);
});
