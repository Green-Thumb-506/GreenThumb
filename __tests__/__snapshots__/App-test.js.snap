// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`API renders the sign-up screen 1`] = `
<RCTScrollView
  contentContainerStyle={
    Object {
      "alignItems": "center",
      "backgroundColor": "#bdc3c7",
      "flex": 1,
      "justifyContent": "flex-start",
      "paddingTop": 50,
    }
  }
>
  <View>
    <Image
      source={1}
      style={
        Object {
          "height": 120,
          "marginBottom": 60,
          "width": 120,
        }
      }
    />
    <View
      style={
        Object {
          "paddingBottom": 10,
          "paddingHorizontal": 10,
          "width": "70%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#86939e",
            "fontSize": 16,
            "fontWeight": "bold",
          }
        }
      >
        Name
      </Text>
      <View
        style={
          Object {
            "alignItems": "center",
            "borderBottomWidth": 1,
            "borderColor": "#86939e",
            "flexDirection": "row",
            "transform": Array [
              Object {
                "translateX": 0,
              },
            ],
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          editable={true}
          onChangeText={[Function]}
          placeholder="John Deer"
          rejectResponderTermination={true}
          style={
            Object {
              "alignSelf": "center",
              "color": "black",
              "flex": 1,
              "fontSize": 18,
              "minHeight": 40,
            }
          }
          testID="RNE__Input__text-input"
          underlineColorAndroid="transparent"
        />
      </View>
    </View>
    <View
      style={
        Object {
          "paddingBottom": 10,
          "paddingHorizontal": 10,
          "width": "70%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#86939e",
            "fontSize": 16,
            "fontWeight": "bold",
          }
        }
      >
        Email
      </Text>
      <View
        style={
          Object {
            "alignItems": "center",
            "borderBottomWidth": 1,
            "borderColor": "#86939e",
            "flexDirection": "row",
            "transform": Array [
              Object {
                "translateX": 0,
              },
            ],
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          editable={true}
          onChangeText={[Function]}
          placeholder="email@address.com"
          rejectResponderTermination={true}
          style={
            Object {
              "alignSelf": "center",
              "color": "black",
              "flex": 1,
              "fontSize": 18,
              "minHeight": 40,
            }
          }
          testID="RNE__Input__text-input"
          underlineColorAndroid="transparent"
        />
      </View>
    </View>
    <View
      style={
        Object {
          "paddingBottom": 20,
          "paddingHorizontal": 10,
          "width": "70%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#86939e",
            "fontSize": 16,
            "fontWeight": "bold",
          }
        }
      >
        Password
      </Text>
      <View
        style={
          Object {
            "alignItems": "center",
            "borderBottomWidth": 1,
            "borderColor": "#86939e",
            "flexDirection": "row",
            "transform": Array [
              Object {
                "translateX": 0,
              },
            ],
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          editable={true}
          onChangeText={[Function]}
          placeholder="Password"
          rejectResponderTermination={true}
          secureTextEntry={true}
          style={
            Object {
              "alignSelf": "center",
              "color": "black",
              "flex": 1,
              "fontSize": 18,
              "minHeight": 40,
            }
          }
          testID="RNE__Input__text-input"
          underlineColorAndroid="transparent"
        />
      </View>
    </View>
    <View
      style={
        Object {
          "borderRadius": 3,
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "color": "#2c3e50",
            "opacity": 1,
            "paddingBottom": 30,
            "width": "70%",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#2c3e50",
              "borderColor": "#2089dc",
              "borderRadius": 3,
              "borderWidth": 0,
              "flexDirection": "row",
              "justifyContent": "center",
              "padding": 8,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 18,
                "paddingBottom": 1,
                "paddingTop": 2,
                "textAlign": "center",
              }
            }
          >
            Create Account
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;

exports[`App renders the loading screen 1`] = `
<AppLoading
  onError={[Function]}
  onFinish={[Function]}
  startAsync={[Function]}
/>
`;

exports[`App renders the loading screen 2`] = `
<AppLoading
  onError={[Function]}
  onFinish={[Function]}
  startAsync={[Function]}
/>
`;

exports[`App renders the loading screen 3`] = `
<AppLoading
  onError={[Function]}
  onFinish={[Function]}
  startAsync={[Function]}
/>
`;

exports[`App renders the root without loading screen 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  <AppNavigator />
</View>
`;

exports[`App renders the root without loading screen 2`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  <AppNavigator />
</View>
`;

exports[`App renders the settings screen 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F5FCFF",
      "flex": 1,
      "justifyContent": "center",
      "position": "absolute",
    }
  }
/>
`;

exports[`App renders the settings screen 2`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F5FCFF",
      "flex": 1,
      "justifyContent": "center",
      "position": "absolute",
    }
  }
/>
`;

exports[`SLibraryScreen renders the settings screen 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F5FCFF",
      "flex": 1,
      "justifyContent": "center",
      "position": "absolute",
    }
  }
/>
`;

exports[`SSignUpScreen renders the sign-up screen 1`] = `
<RCTScrollView
  contentContainerStyle={
    Object {
      "alignItems": "center",
      "backgroundColor": "#bdc3c7",
      "flex": 1,
      "justifyContent": "flex-start",
      "paddingTop": 50,
    }
  }
>
  <View>
    <Image
      source={1}
      style={
        Object {
          "height": 120,
          "marginBottom": 60,
          "width": 120,
        }
      }
    />
    <View
      style={
        Object {
          "paddingBottom": 10,
          "paddingHorizontal": 10,
          "width": "70%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#86939e",
            "fontSize": 16,
            "fontWeight": "bold",
          }
        }
      >
        Name
      </Text>
      <View
        style={
          Object {
            "alignItems": "center",
            "borderBottomWidth": 1,
            "borderColor": "#86939e",
            "flexDirection": "row",
            "transform": Array [
              Object {
                "translateX": 0,
              },
            ],
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          editable={true}
          onChangeText={[Function]}
          placeholder="John Deer"
          rejectResponderTermination={true}
          style={
            Object {
              "alignSelf": "center",
              "color": "black",
              "flex": 1,
              "fontSize": 18,
              "minHeight": 40,
            }
          }
          testID="RNE__Input__text-input"
          underlineColorAndroid="transparent"
        />
      </View>
    </View>
    <View
      style={
        Object {
          "paddingBottom": 10,
          "paddingHorizontal": 10,
          "width": "70%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#86939e",
            "fontSize": 16,
            "fontWeight": "bold",
          }
        }
      >
        Email
      </Text>
      <View
        style={
          Object {
            "alignItems": "center",
            "borderBottomWidth": 1,
            "borderColor": "#86939e",
            "flexDirection": "row",
            "transform": Array [
              Object {
                "translateX": 0,
              },
            ],
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          editable={true}
          onChangeText={[Function]}
          placeholder="email@address.com"
          rejectResponderTermination={true}
          style={
            Object {
              "alignSelf": "center",
              "color": "black",
              "flex": 1,
              "fontSize": 18,
              "minHeight": 40,
            }
          }
          testID="RNE__Input__text-input"
          underlineColorAndroid="transparent"
        />
      </View>
    </View>
    <View
      style={
        Object {
          "paddingBottom": 20,
          "paddingHorizontal": 10,
          "width": "70%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#86939e",
            "fontSize": 16,
            "fontWeight": "bold",
          }
        }
      >
        Password
      </Text>
      <View
        style={
          Object {
            "alignItems": "center",
            "borderBottomWidth": 1,
            "borderColor": "#86939e",
            "flexDirection": "row",
            "transform": Array [
              Object {
                "translateX": 0,
              },
            ],
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          editable={true}
          onChangeText={[Function]}
          placeholder="Password"
          rejectResponderTermination={true}
          secureTextEntry={true}
          style={
            Object {
              "alignSelf": "center",
              "color": "black",
              "flex": 1,
              "fontSize": 18,
              "minHeight": 40,
            }
          }
          testID="RNE__Input__text-input"
          underlineColorAndroid="transparent"
        />
      </View>
    </View>
    <View
      style={
        Object {
          "borderRadius": 3,
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "color": "#2c3e50",
            "opacity": 1,
            "paddingBottom": 30,
            "width": "70%",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#2c3e50",
              "borderColor": "#2089dc",
              "borderRadius": 3,
              "borderWidth": 0,
              "flexDirection": "row",
              "justifyContent": "center",
              "padding": 8,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 18,
                "paddingBottom": 1,
                "paddingTop": 2,
                "textAlign": "center",
              }
            }
          >
            Create Account
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
